extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeStreamExtractor.java

```diff
@@ -810,6 +811,8 @@ public String getErrorMessage() {
             "\\bc\\s*&&\\s*d\\.set\\([^,]+\\s*,\\s*(:encodeURIComponent\\s*\\()([a-zA-Z0-9$]+)\\("
     };
     private static final String STS_REGEX = "signatureTimestamp[=:](\\d+)";
+    private static final String AUDIO_STREAM_TYPE_REGEX =
```

Comment:
Due to the number of executions of this code, please compile the regex string as a `Pattern` and pass directly this pattern.

This would increase by a lot the performance of your current code, especially when it will run on servers such as Piped backends.